rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMPage.java

```diff
@@ -574,4 +555,28 @@ public void logCriticalMessage(String message) {
         this.lastCriticalMessage = System.currentTimeMillis();
         this.errorButton.show();
     }
+
+    public void showNodeSourceCreationWindow() {
+        destroyNodeSourceWindow();
+        RMPage.this.nsWindow = new NodeSourceCreationWindow(controller);
+        RMPage.this.nsWindow.show();
+    }
+
+    public void showNodeSourceEditWindow(String nodeSourceName) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "logMessage" on line 557. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)